js/ui/status/keyboard.js

```diff
@@ -525,7 +575,7 @@ const InputSourceManager = new Lang.Class({
         this._mruSources = mruSources.concat(sourcesList);
     },
 
-    _inputSourcesChanged: function() {
+    _inputSourcesChanged: function(needsFallbackSource = false) {
```

Comment:
In certain cases, `_inputSourcesChanged()` is called as a callback for settings changed signals. In that case, the first parameter here will be the settings object itself. I think to avoid churn, it would be cleanest if `inputSourcesChanged()` was left without additional parameters, and its body split into e.g. `_updateInputSources()` that has the extra parameter. `_syncPasswordMode()` would then just call `_updateInputSources()`.

Reply:
You are absolutely right. Side-note: It would be cleaner if those callbacks would list the arguments and if unused do prepend with a '_' or similar to make linters happy. Not starting to think about named arguments e.g. Gtk.ComboBox(model=self.auth_store)...Sigh :)

Anyhow, adjusted accordingly.